src/overlays/actors/ovl_En_Racedog/z_en_racedog.c

```diff
@@ -52,31 +90,65 @@ const ActorInit En_Racedog_InitVars = {
+ * - At the very start of the race, all dogs will use the first value for their color, but only for a single point
+ *   along the race track's path. After that, they will have a base speed of 5.0, regardless of what's in the table
+ *   for their color. The sole exception to this is the blue dog, who will use its first value for the entire
+ *   first 1/4th of the race.
+ * - For the last 3/4ths of the race, all dogs will check to see if their current point along the race track path is
+ *   greater than or equal to the pointToUseSecondBaseSpeed for their RaceDogInfo. If it is, then they will use the
+ *   second value for their color as their base speed; otherwise, they will use 5.0.
+ */
+static f32 sBaseSpeeds[][2] = {
     { 0.0f, 0.0f }, { 5.0f, 5.5f }, { 5.0f, 5.0f }, { 5.5f, 5.0f }, { 4.5f, 5.5f }, { 6.0f, 4.0f }, { 4.0f, 6.0f },
 };
 
-static UnkRacedogStruct D_80B25D88[] = {
-    { -1.0f, 1.20000004768f, 3, 0, 9, 0x3539 },  { -1.0f, 1.20000004768f, 1, 1, 9, 0x353A },
-    { -1.0f, 1.20000004768f, 5, 2, 10, 0x353B }, { -1.0f, 1.20000004768f, 2, 3, 9, 0x353C },
-    { -1.0f, 1.20000004768f, 4, 4, 8, 0x353D },  { -1.0f, 1.20000004768f, 2, 5, 9, 0x353E },
-    { -1.0f, 1.20000004768f, 3, 6, 9, 0x353F },  { -1.0f, 1.20000004768f, 1, 7, 9, 0x3540 },
-    { -1.0f, 1.20000004768f, 1, 8, 9, 0x3541 },  { -1.0f, 1.20000004768f, 6, 9, 8, 0x3542 },
-    { -1.0f, 1.20000004768f, 2, 10, 9, 0x3543 }, { -1.0f, 1.20000004768f, 3, 11, 9, 0x3544 },
-    { -1.0f, 1.20000004768f, 1, 12, 9, 0x3545 }, { -1.0f, 1.20000004768f, 4, 13, 8, 0x3546 },
+/**
+ * A table of RaceDogInfo for every dog in the race. Note that the sprintSpeedMultiplier and
+ * textId values in this table are updated by functions within this actor.
+ */
+static RaceDogInfo sDogInfo[] = {
+    { -1.0f, 1.2f, DOG_COLOR_BEIGE, 0, 9, 0x3539 },  { -1.0f, 1.2f, DOG_COLOR_WHITE, 1, 9, 0x353A },
+    { -1.0f, 1.2f, DOG_COLOR_BLUE, 2, 10, 0x353B },  { -1.0f, 1.2f, DOG_COLOR_GRAY, 3, 9, 0x353C },
+    { -1.0f, 1.2f, DOG_COLOR_BROWN, 4, 8, 0x353D },  { -1.0f, 1.2f, DOG_COLOR_GRAY, 5, 9, 0x353E },
+    { -1.0f, 1.2f, DOG_COLOR_BEIGE, 6, 9, 0x353F },  { -1.0f, 1.2f, DOG_COLOR_WHITE, 7, 9, 0x3540 },
+    { -1.0f, 1.2f, DOG_COLOR_WHITE, 8, 9, 0x3541 },  { -1.0f, 1.2f, DOG_COLOR_GOLD, 9, 8, 0x3542 },
+    { -1.0f, 1.2f, DOG_COLOR_GRAY, 10, 9, 0x3543 },  { -1.0f, 1.2f, DOG_COLOR_BEIGE, 11, 9, 0x3544 },
+    { -1.0f, 1.2f, DOG_COLOR_WHITE, 12, 9, 0x3545 }, { -1.0f, 1.2f, DOG_COLOR_BROWN, 13, 8, 0x3546 },
 };
 
-static UnkRacedogStruct D_80B25E68 = { -1.0f, 1.0, 0, -1, 0, 0x353E };
+/**
+ * Stores the RacedogInfo for the dog that is selected by the player. These values are just
+ * placeholders, and the actual value gets grabbed from sDogInfo in EnRacedog_Init.
+ */
+static RaceDogInfo sSelectedDogInfo = { -1.0f, 1.0, DOG_COLOR_DEFAULT, -1, 0, 0x353E };
 
-static Vec2f D_80B25E78[] = {
+/**
+ * The XZ-coordinates used to determine if the dog is inside the finish line.
```

Comment:
Definitely inside. `EnRacedog_IsOverFinishLine` is basically using cross products (technically edge products since it's 2D, but whatever) to check if the dog's position is within the area defined by these four points.